samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -0,0 +1,164 @@
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceNodeLabel(const char *label, size_t labelLength,
+						       uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Store(&id, label, labelLength);
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceNodeLabel(char *label, size_t labelMaxLength, size_t &labelLength,
+						      uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Load(&id, label, labelMaxLength, labelLength);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceNodeLabel(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceType(uint16_t deviceType, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return PersistentStorage::Instance().Store(&id, &deviceType, sizeof(deviceType));
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceType(uint16_t &deviceType, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return LoadSimpleData(&id, deviceType);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceType(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDevice(BridgedDevice *device, uint8_t index)
+{
+	if (!device) {
+		return false;
+	}
+
+	if (!StoreBridgedDeviceEndpointId(static_cast<uint16_t>(device->GetEndpointId()), index)) {
+		return false;
+	}
+
+	if (!StoreBridgedDeviceNodeLabel(device->GetNodeLabel(), strlen(device->GetNodeLabel()), index)) {
+		return false;
+	}
+
+	return StoreBridgedDeviceType(static_cast<uint16_t>(device->GetDeviceType()), index);
+}
+
+#ifdef CONFIG_BRIDGED_DEVICE_BT
+bool BridgeStorageManager::StoreBtAddress(bt_addr_le_t addr, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBtAddress);
+
+	return PersistentStorage::Instance().Store(&id, &addr, sizeof(addr));
+}
+
+bool BridgeStorageManager::LoadBtAddress(bt_addr_le_t &addr, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBtAddress);
+
+	return LoadSimpleData(&id, addr);
+}
+
+bool BridgeStorageManager::RemoveBtAddress(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBtAddress);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
```

Comment:
Paradoxically saving settings space is a reason why I haven't used BT settings :) They are much heavier, than what we currently have and contain multiple information that we are not going to use, so probably it would be reasonable to not use them at all and only include small subset of necessary information in our space.

Reply:
ok that convinces me :)